backend/app/schemas/crop.py

```
@@ -0,0 +1,65 @@
+    @field_validator("crop_type")
+    @classmethod
+    def validate_crop_type(cls, v: str) -> str:
+        v = v.lower().strip()
+        if v not in COMMON_CROPS:
+            raise ValueError(f"crop_type must be one of {sorted(COMMON_CROPS)}")
+        return v
+
+    @model_validator(mode="after")
+    def validate_other(self) -> "CropRegistrationCreate":
+        if self.crop_type == "other" and not self.crop_type_other:
+            raise ValueError("crop_type_other is required when crop_type is 'other'")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)